suite.go

```diff
@@ -0,0 +1,110 @@
+// Licensed under the MIT license, see LICENCE file for details.
+
+package quicktest
+
+import (
+	"reflect"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+)
+
+// RunSuite runs all the tests defined in the given suite as a separate
+// subtest. A test is a method on the suite of the form
+//	func (T) TestXxx(*quicktest.C)
+// where Xxx does not start with a lowercase letter.
+//
+// If suite is is a pointer, the value pointed to is copied
+// before any methods are invoked on it; a new copy
+// is made for each test.
+//
+// If suite has a method of the form
+//	func (T) Init(*quicktest.C)
+// this method will be invoked before each test runs.
+//
+// For instance
+//	type testSuite struct{}
+//
+//	func (testSuite) Init(c *qt.C) {
```

Comment:
I'd make this example simpler, and it would be nice if it didn't modify global state.
Also, it should probably be an actual Go example rather than just in the comment.